sswlib/src/main/java/components/BipedLoadout.java

```diff
@@ -1653,6 +1653,76 @@ public ArrayList FindIndexes( abPlaceable p ) {
         return v;
     }
 
+    public List<LocationIndex> FindIndexesByName(String lookupName) {
+        List<LocationIndex> locations = new ArrayList<>();
+        for (int i = 0; i < 6; i++) {
+            if (HDCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_HD));
+            }
+            if (CTCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_CT));
+            }
+            if (LTCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_LT));
+            }
+            if (RTCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_RT));
+            }
+            if (LACrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_LA));
+            }
+            if (RACrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_RA));
+            }
+            if (LLCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_LL));
+            }
+            if (RLCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_RL));
+            }
+        }
+        for (int i = 6; i < 12; i++) {
+            if (CTCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_CT));
+            }
+            if (LTCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_LT));
+            }
+            if (RTCrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_RT));
+            }
+            if (LACrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_LA));
+            }
+            if (RACrits[i].LookupName().equals(lookupName)) {
+                locations.add(new LocationIndex(i, LocationIndex.MECH_LOC_RA));
+            }
+        }
+        return locations;
+    }
+
+    public boolean HasHarjel() {

```

Comment:
I kind of hate how these methods are called "HasHarjel" but only check for harjel 2 and 3. The issue was that harjel 1 isn't relevant for basically anything, whereas 2/3 apply different armor type modifiers. I just couldn't think of a name that covered both HarJel 2 and 3, and I didn't want to split into two functions.

Reply:
I would go off of what it is later used for.  I'm not at my dev system and can't find where this is consumed at the moment, but I assume it's for calculation or restriction purposes?

Reply:
Yep, I'm looking at renaming it to "HasHarjelMod" now, since that more clearly indicates that it's checking for the presence of a harjel armor mod, likewise with the second version that checks by location.

Reply:
That's a good change.